README.md

```diff
@@ -91,97 +107,173 @@ cloning over SSH may provide a better experience (which requires
     git clone git@github.com:apple/swift.git
     ./swift/utils/update-checkout --clone-with-ssh
 
-#### CMake
-[CMake](https://cmake.org) is the core infrastructure used to configure builds of 
-Swift and its companion projects; at least version 3.4.3 is required. 
+### Building Swift
 
-On macOS, you can download the [CMake Binary Distribution](https://cmake.org/download),
-bundled as an application, copy it to `/Applications`, and add the embedded
-command line tools to your `PATH`:
+The `build-script` is a high-level build automation script that supports basic
+options such as building a Swift-compatible LLDB, building the Swift Package
+Manager, building for various platforms, running tests after builds, and more.
 
-    export PATH=/Applications/CMake.app/Contents/bin:$PATH
-    
-On Linux, if you have not already installed Swift's [development
-dependencies](#linux-dependencies), you can download and install the CMake
-package separately using the following command:
+View the inline help to find out more, in particular the section "Typical uses":
 
-    sudo apt-get install cmake
-    
+    utils/build-script -h
 
-#### Ninja
-[Ninja](https://ninja-build.org) is the current recommended build system
-for building Swift and is the default configuration generated by CMake. [Pre-built
-packages](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-are available for macOS and Linux distributions. You can also clone Ninja
-next to the other projects and it will be bootstrapped automatically:
+There are two primary build options to use with CMake, xcodebuild or Ninja.
+xcodebuild integrates with Xcode.app, but Ninja is a bit faster and
+supports more environments.
 
-**Via HTTPS**
+To build using Ninja, run:
 
-    git clone https://github.com/ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+    utils/build-script --release-debuginfo
 
-**Via SSH**
+If you are interested in compiler development it helps to build LLVM + Clang
+with optimizations enabled, but enable debug support in Swift. This is a good
+starting point since the debugger will work in the swift code base, and the
+tests will run faster:
 
-    git clone git@github.com:ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+    utils/build-script --release-debuginfo --debug-swift
 
-#### Install via third-party packaging tool (macOS only)
+This will only build the compiler, and it will not build the standard library.
```

Comment:
This makes it sound like you will not build the standard library rather than what will happen (you will build the stdlib but you will suffer through building an optimized stdlib with a debug compiler).